ocw/lib/EC2.py

```diff
@@ -85,11 +86,14 @@ def all_regions(self):
         return regions
 
     def delete_instance(self, instance_id):
-        instances_list = list(self.ec2_resource().instances.filter(InstanceIds=[instance_id]))
-        if len(instances_list) > 0:
-            instances_list[0].terminate()
-        else:
-            logger.warning("Instance {} is ACTIVE in local DB but does not exists on EC2".format(instance_id))
+        try:
+            self.ec2_resource().instances.filter(InstanceIds=[instance_id]).terminate()
```

Comment:
so, this doesn't return a list?

Reply:
no it returns something which suppose to be converted the list ( `instanceCollection` ) but it is not the point of this PR . 
Before my assumption was that exception happens when object returned by `filter` trying to call terminate. But it appears that `filter` function itself is main cause . It throwing exception when you try to filter by non-existing entity instead of just return empty collection